packages/plugin-commands-installation/test/installDeep.ts

```diff
@@ -0,0 +1,67 @@
+/// <reference path="../../../typings/index.d.ts" />
+import assertProject from '@pnpm/assert-project'
+import { install } from '@pnpm/plugin-commands-installation'
+import { readProjects } from '@pnpm/filter-workspace-packages'
+import fixtures from '@pnpm/test-fixtures'
+import rimraf from '@zkochan/rimraf'
+import path from 'path'
+import { promises as fs } from 'fs'
+import { DEFAULT_OPTS } from './utils'
+
+const f = fixtures(__dirname)
+
+test('Installing a package deeply installs all required dependencies', async () => {
+  f.prepare('workspace-external-depends-deep')
+  const cwd = process.cwd()
+  const {
+    allProjects,
+    selectedProjectsGraph,
+  } = await readProjects(process.cwd(), [
+    { namePattern: '@kenrick95/internal-g' },
+  ])
+  for (const project of allProjects) {
+    await rimraf(path.join(project.dir, 'node_modules'))
+  }
+
+  const projectGraphKeys = Object.keys(selectedProjectsGraph)
+  expect(projectGraphKeys).toHaveLength(1)
+  expect(selectedProjectsGraph[projectGraphKeys[0]].package.manifest.name).toBe(
+    '@kenrick95/internal-g'
+  )
+
+  console.log('projectGraphKeys', projectGraphKeys)
+  console.log('cwd', cwd)
+
+  await install.handler({
+    ...DEFAULT_OPTS,
+    dir: cwd,
+    workspaceDir: cwd,
+
+    allProjects,
+    recursive: true,
+    selectedProjectsGraph,
+    preferWorkspacePackages: true,
+    linkWorkspacePackages: 'deep',
+  })
+
+  for (const project of allProjects) {
+    const files = await fs.readdir(project.dir)
+    console.log(project.dir, files)
+    if (files.includes('node_modules')) {
+      console.log(
+        project.dir,
+        'node_modules',
+        await fs.readdir(path.join(project.dir, 'node_modules'))
+      )
+      // Somehow "g" now has pnpm-lock.yaml ???
```

Comment:
From the logs, I found that this results in this layout... Feels like I have some wrong parameter when calling `install.handler` but not sure what. My intention is to run the equivalent of `pnpm install --filter "@kenrick95/internal-g"`


```
root
  .npmrc
  package.json
  packages
  pnpm-lock.yaml
  pnpm-workspace.yaml
  f
    package.json
    node_modules
      is-positive
      is-negative
  g
    package.json
    pnpm-lock.yaml
    node_modules
      .modules.yaml
      .pnpm
      is-positive
      is-negative
```